### ISSUE TITLE
ALC1220 distorted sound in front green jack

### ISSUE THREAD

**Issue by johnzaro** [toleda/audio_CloverALC#114]
I have installed the latest version of appleALC in which you have recently added audio id's 1 and 2 for the ALC1220. With audio I'd 1 the green back jack outputs audio without any issue and also the front pink jack is correctly recognized and  working as a microphone. However the front green jack while being correctly recognized as headphones and auto switches between back and front green jack when I connect headphones , it outputs a very distorted and broken sound especially in lower frequencies of sound.
 Is there a way to fix this small issue as everything else is working perfectly fine?

**Comment by johnzaro** [comment]
I have a gigabyte Z270N-WiFi so it has the ALC1220 and a Bitfenix Prodigy case.

I tried using the audio id 5 from Mirone and if I unplug the back green jack for speakers and connect the front green jack for headphones (it does not support auto switch) then the audio is playing fine through headphones.

Is there anything that can be done for audio to work through headphones with Id 1 of yours as with this id I am able to auto switch where to output sound and also use my headphones mic with front pink jack?

**Comment by toleda** [comment]
Support for 1220/Audio IDs: 1 or 2 only.

**Comment by johnzaro** [comment]
Support for ID 1 is what I am asking for

**Comment by toleda** [comment]
1220/Audio ID: 1 and 2/Headphone switching supported. 
If Headphones/Front/Green connected and SysPref/Sound/Output says Internal Speakers; hardware problem, connector or cable.

**Comment by malhal** [comment]
Same problem here with Corsair 330R. I think it probably is a hardware issue as I read on windows forums people having to RMA the front ports because of grounding issues.

I'll be testing audio_CloverALC 1220 and if successful I can update the Readme adding 1220 to the list of supported chips.

**Comment by toleda** [comment]
ALC1220, both versions supported.

**Comment by malhal** [comment]
Yep was successful on my hardware, submitted PR to update Readme.

**Comment by johnzaro** [comment]
What do you mean?? Is there a way for the front audio to work correctly?

**Comment by malhal** [comment]
Sorry I hijacked your issue, I'll keep investigating the front audio because no it's not fixed, it is still garbled. If anyone else gets this issue made they could post their motherboard and case here too.

**Comment by johnzaro** [comment]
I don't think it is a hardware issue as I mentioned above that I tried using the audio id 5 from Mirone in AppleALC (which does not support auto switch between back and front green jack) so if I unplug the back green jack for speakers and connect the front green jack for headphones then the audio is playing fine through headphones. Also in windows front green jack also works fine.

**Comment by barrrrt** [comment]
I have the same problem with crackling sound with the front green jack.
I use a Asus Z170 Pro Gaming (so its a 1150). Still no luck to solve the issue.

**Comment by toleda** [comment]
1150 off topic.  
Distortion/crackling is a user system specific hardware problem. Front panel audio issues are a case/connector problem.

**Comment by barrrrt** [comment]
Yes, i know its a different chip. But the problem seems to be the same. @toleda do you think a driver from windows can change stuff, so that the problem occurs like described here?
I just booted back to windows and back to OSX, and the problem came back.

**Comment by toleda** [comment]
Title of thread is 1220/front panel distortion; 1150/Win 10 ???
Win 10 does not cause macOS audio distortion
Win 10 can cause macOS no audio

**Comment by joseracer** [comment]
Hi Toleda, thanks for your hard work.
I´m experiencing the same problem (front headphone´s jack distortion.) on a Gigabyte Z270x-Gaming 5, and had the same on a Gigabyte Z270N-Gaming 5 before this one. I´ve found that the problem is only when the OS volume level is >30% only, seems this can be the Gigabyte included headphone AMP causing the issue. Will probably try a fresh install later on to see if that fixes it, but at the lower OS volume settings the distortion is not there and volume is actually loud enough, just thought it might help sharing the issue here.

Thanks again!

**Comment by toleda** [comment]
macOS/Desktop/AppleHDA does not support external amps. Try https://github.com/RehabMan/EAPD-Codec-Commander

**Comment by lobmarib** [comment]
I know this is an old topic, but any progress on this? I'm experiencing the same issue (same mobo) running mojave (AppleALC only as codec commander was discontinued)

**Comment by malhal** [comment]
I've just been setting volume level to 1 which is still very loud and distorted but by setting volume in the app to medium and it limits most of the distortion.
